tests/importer/importer.cpp

```diff
@@ -0,0 +1,766 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License->
+You may obtain a copy of the License at
+
+    http://www.apache->org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License->
+*/
+
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+#include "test_utils.h"
+
+TEST(Importer, create)
+{
+    auto importer = libcellml::Importer::create();
+    EXPECT_NE(nullptr, importer);
+}
+
+TEST(Importer, noWarningForkedImport)
+{
+    libcellml::ParserPtr p = libcellml::Parser::create();
+    libcellml::ImporterPtr importer = libcellml::Importer::create();
+    libcellml::ModelPtr model = p->parseModel(fileContents("resolveimports/forkedImport.cellml"));
+
+    EXPECT_EQ(size_t(0), p->errorCount());
+
+    EXPECT_TRUE(model->hasUnresolvedImports());
+    importer->resolveImports(model, resourcePath("resolveimports/"));
+    EXPECT_FALSE(model->hasUnresolvedImports());
+}
+
+TEST(Importer, noWarningDiamondImport)
+{
+    libcellml::ParserPtr p = libcellml::Parser::create();
+    libcellml::ImporterPtr importer = libcellml::Importer::create();
+    libcellml::ModelPtr model = p->parseModel(fileContents("resolveimports/diamond.cellml"));
+
+    EXPECT_EQ(size_t(0), p->errorCount());
+
+    EXPECT_TRUE(model->hasUnresolvedImports());
+    importer->resolveImports(model, resourcePath("resolveimports/"));
+    EXPECT_FALSE(model->hasUnresolvedImports());
+}
+
+TEST(Importer, warningCircularImportReferencesComponent)
+{
+    std::string warningMessage = "Cyclic dependencies were found when attempting to resolve components in model 'circularImport1'. The dependency loop is:\n"
+                                 "    - component 'i_am_cyclic' imports 'c2' from 'circularImport_2.cellml';\n"
+                                 "    - component 'c2' imports 'c3' from 'circularImport_3.cellml';\n"
+                                 "    - component 'c3' imports 'i_am_cyclic' from 'circularImport_1.cellml';\n"
+                                 "    - component 'i_am_cyclic' imports 'c2' from 'circularImport_2.cellml'.";
```

Comment:
Right now, the warning message reads:
```
Cyclic dependencies were found when attempting to resolve components in model 'circularImport1'. The dependency loop is:
    - component 'i_am_cyclic' imports 'c2' from 'circularImport_2.cellml';
    - component 'c2' imports 'c3' from 'circularImport_3.cellml';
    - component 'c3' imports 'i_am_cyclic' from 'circularImport_1.cellml';
    - component 'i_am_cyclic' imports 'c2' from 'circularImport_2.cellml'.
```

but I was really after:

```
Cyclic dependencies were found when attempting to resolve components in model 'circularImport1'. The dependency loop is:
 - component 'i_am_cyclic' imports 'c2' from 'circularImport_2.cellml';
 - component 'c2' imports 'c3' from 'circularImport_3.cellml';
 - component 'c3' imports 'i_am_cyclic' from 'circularImport_1.cellml'; and
 - component 'i_am_cyclic' imports 'c2' from 'circularImport_2.cellml'.
```

Reply:
Otherwise, rather than:
```
component 'xxx' import 'yyy' from 'zzz.cellml'
```
I would have:
```
component 'xxx' is imported from 'yyy' in 'zzz.cellml'
```
Indeed, `xxx` doesn't import `yyy`, but it is imported from `yyy`.

Reply:
a6280ae3